core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -612,7 +612,7 @@ private boolean rowTypesAreEquivalent(
       return litmus.fail("Mismatch for column count: [{}]", Pair.of(rel0, rel1));
     }
     for (Pair<RelDataTypeField, RelDataTypeField> pair
-        : Pair.zip(rel0.rowType.getFieldList(), rel0.rowType.getFieldList())) {
+        : Pair.zip(rel0.rowType.getFieldList(), rel1.rowType.getFieldList())) {
       if (!pair.left.getType().equals(pair.right.getType())) {
         return litmus.fail("Mismatch for column type: [{}]", Pair.of(rel0, rel1));
```

Comment:
Can you give a test case, thanks.

Reply:
Danny~
We should always have test when proposing change, especially when modifying a function for assertion.
But it might be hard for this chagne.
The existing code asserts `rel0` and `rel0` has the equivalent row type by typo mistake. But what it really means is that `rel0` and `rel1` have the equivalent row type, which is guaranteed by matching logic of `SubstitutionVisitor`. As we can see all the tests in MaterializationTest.java passed now.
What I want to say is that it might be hard to have a light way to construct a test which fails without this PR but passes with this PR.
But I'm happy if @cndaimin have an good approach to test it.

Reply:
IMO, it is not very hard to provide a test case for this change since the code before change will never throw an exception due to `pair.left.getType().equals(pair.right.getType())` is always true.

Reply:
OK, I will figure out how to add a test on it. And, sorry for the late reply~